transformations/diverse_paraphrase/submod/submodopt.py

```diff
@@ -0,0 +1,81 @@
+import numpy as np
+import scipy.linalg as la
+import pdb
+
+from transformations.diverse_paraphrase.submod.submodular_funcs import (
+    distinct_ngrams,
+    ngram_overlap,
+    similarity_func,
+    seq_func,
+    ngram_overlap_unit,
+    similarity_gain,
+    seq_gain,
+)
+
+
+class SubmodularOpt:
+    def __init__(self, V=None, v=None, **kwargs):
+        self.v = v
+        self.V = V
+
+    def initialize_function(self, lam, a1=1.0, a2=1.0, b1=1.0, b2=1.0):
```

Comment:
It would be good to add some comments about what these arguments are.

Reply:
I'll add the comments